public/scripts/api-helpers.js

```diff
@@ -131,4 +117,71 @@ const callAPIByCategory = async (task) => {
     default:
       return task;
   }
-}
+};
+
+/**
+ * Gives a time limit to a promise
+ * Given promise is rejected if not resolved before timeout
+ * @param  {number}      ms      Length of ms before timeout
+ * @param  {Promise<{}>} promise Promise to limit
+ * @return {Promise<>}
+ */
+const timeoutPromise = (ms, promise) => {
+  const timeout = new Promise((resolve, reject) =>
+    setTimeout(() =>
+      reject(`Timed out after ${ms} ms.`), ms));
+
+  return Promise.race([
+    promise,
+    timeout
+  ]);
+};
+
+/**
+ * Checks if given word is found in the given string
+ * @param  {string}  str  String to check condition
+ * @param  {string}  word Word to look for in the string
+ * @return {boolean}      Boolean if word found in string
+ */
+const findWordInString = (str, word) => {
+  return new RegExp('\\b' + word + '\\b', 'i').test(str);
+};
+
+/**
+ * Matches string against key values to check for easy API match
+ * @param  {string} string Takes in string to check for keyword
+ * @return {string}        String of category matched, else undefined
+ */
+const matchCategoryKeyword = (string) => {
+  let category;
+  const categories = ['restaurants', 'films', 'books', 'products'];
+  const keywords = ['eat', 'watch', 'read', 'buy'];
+
+  keywords.some((val, index) => {
```

Comment:
Potential problem: if the title of a movie/book/product/restaurants contains any of the keywords whose index is larger than the actual keyword, it overwrites the correct category.
 For example "Watch The Reader", because the callback doesn't have a return value, it never returns true and runs the callback for each of the keywords, so when it  continues running the callback after it finds "Watch" and will change category to "read".

Reply:
Suggestion: search through the string from start to end (rather than through the keywords array from start to end) and start at first match because the actual keyword is usually before the movie title, book title or restaurant name etc.